velox/exec/fuzzer/RowNumberFuzzerBase.cpp

```diff
@@ -0,0 +1,286 @@
+      "Either --steps or --duration_sec needs to be greater than zero.");
+  VELOX_USER_CHECK_GE(FLAGS_batch_size, 10, "Batch size must be at least 10.");
+
+  const auto startTime = std::chrono::system_clock::now();
+  size_t iteration = 0;
+
+  while (!isDone(iteration, startTime)) {
+    LOG(INFO) << "==============================> Started iteration "
+              << iteration << " (seed: " << currentSeed_ << ")";
+    runSingleIteration();
+    LOG(INFO) << "==============================> Done with iteration "
+              << iteration;
+
+    reSeed();
+    ++iteration;
+  }
+}
+
+RowVectorPtr RowNumberFuzzerBase::execute(
+    const PlanWithSplits& plan,
+    const std::shared_ptr<memory::MemoryPool>& pool,
+    bool injectSpill,
+    bool injectOOM,
+    const std::optional<std::string>& spillConfig,
+    int maxSpillLevel) {
+  LOG(INFO) << "Executing query plan: " << plan.plan->toString(true, true);
+
+  test::AssertQueryBuilder builder(plan.plan);
+  if (!plan.splits.empty()) {
+    builder.splits(plan.splits);
+  }
+
+  int32_t spillPct{0};
+  if (injectSpill) {
+    VELOX_CHECK(
+        spillConfig.has_value(),
+        "Spill config not set for execute with spilling");
+    VELOX_CHECK_GE(
+        maxSpillLevel, 0, "Max spill should be set for execute with spilling");
+    std::shared_ptr<test::TempDirectoryPath> spillDirectory;
+    spillDirectory = exec::test::TempDirectoryPath::create();
+    builder.config(core::QueryConfig::kSpillEnabled, true)
+        .config(core::QueryConfig::kMaxSpillLevel, maxSpillLevel)
+        .config(spillConfig.value(), true)
+        .spillDirectory(spillDirectory->getPath());
+    spillPct = 10;
+  }
+
+  test::ScopedOOMInjector oomInjector(
+      []() -> bool { return folly::Random::oneIn(10); },
+      10); // Check the condition every 10 ms.
+  if (injectOOM) {
+    oomInjector.enable();
+  }
+
+  // Wait for the task to be destroyed before start next query execution to
+  // avoid the potential interference of the background activities across query
+  // executions.
+  auto stopGuard =
+      folly::makeGuard([&]() { test::waitForAllTasksToBeDeleted(); });
+
+  TestScopedSpillInjection scopedSpillInjection(spillPct);
+  RowVectorPtr result;
+  try {
+    result = builder.copyResults(pool.get());
+  } catch (VeloxRuntimeError& e) {
+    if (injectOOM &&
+        e.errorCode() == facebook::velox::error_code::kMemCapExceeded &&
+        e.message() == test::ScopedOOMInjector::kErrorMessage) {
+      // If we enabled OOM injection we expect the exception thrown by the
+      // ScopedOOMInjector.
+      return nullptr;
+    }
+
+    throw e;
+  }
+
+  if (VLOG_IS_ON(1)) {
+    VLOG(1) << std::endl << result->toString(0, result->size());
+  }
+
+  return result;
+}
+
+void RowNumberFuzzerBase::testPlan(
+    const PlanWithSplits& plan,
+    int32_t testNumber,
+    const RowVectorPtr& expected,
+    const std::shared_ptr<memory::MemoryPool>& pool,
```

Comment:
nit: This `pool` argument is not used and can be removed.

Reply:
Removed.